tools/lsp/preview/drop_location.rs

```diff
@@ -546,6 +546,75 @@ fn find_element_to_drop_into(
     tmp
 }
 
+fn can_drop_into(
+    selected_element: &common::ElementRcNode,
+    drop_target: &common::ElementRcNode,
+) -> bool {
+    // Not reparenting, so this most likely will work ;-)
+    if drop_target.children().contains(selected_element) {
+        return true;
+    }
```

Comment:
Basically, what you can check, is if they are in the same scope instead of the same parent. Either from the nodes: try to find the first ancestor which is either a Component, or a ConditionalElement; or a RepeatedElement, and they must be the same for both node.
Or easier from the ElementRc if they are in the same item tree (The enclosing component of the ElementRc)